### PR TITLE
Move Air Two lookup to s3

### LINKED ISSUE #79: Move air two lookup table to s3

### PR DESCRIPTION
**Pull request by EFT-Defra**
Closes #79 by uploading the lookup CSV to S3 and updating the functions that depend on it to read it from there.

### CODE REVIEW

File: src/oiflib/air/two/enrich.py [hunk 1]
```diff
@@ -3,30 +3,55 @@
...
+        filepath_or_buffer=path_to_lookup,
     )
 
     return merge(
         left=df,
-        right=df_lookup,
+        right=_lookup,
         how="left",
         on=["NCFormat", "IPCC"],
     )
```
Comment (EFT-Defra) [inline src/oiflib/air/two/enrich.py:39]: 1. I moved the path_to_lookup from a hard coded value within the function body to a parameter that's passed to the function
2. I added a private function note and example to the doc string

File: src/oiflib/air/two/enrich.py [hunk 2]
```diff
@@ -3,30 +3,55 @@
...
 
 
-def agg_CO2e_by_category_and_year(df: DataFrame) -> DataFrame:
+def _agg_CO2e_by_category_and_year(df: DataFrame) -> DataFrame:
```
Comment (EFT-Defra) [inline src/oiflib/air/two/enrich.py:42]: Added an underscore to the function name. This is a Python convention that indicates that the function is private, i.e. it shouldn't be called directly. It doesn't actually do anything but I think it makes my intentions clearer.

File: src/oiflib/air/two/enrich.py [hunk 3]
```diff
@@ -3,30 +3,55 @@
...
+    .. note:
+        This is a private function. It is not intended to be called directly.
+        It is called within :func:`enrich_air_two`.
+
+    Example:
+        >>> _aggregated: DataFrame = _agg_CO2e_by_category_and_year(
+            df=_joined,
+        )
+
+
```
Comment (EFT-Defra) [inline src/oiflib/air/two/enrich.py:54]: Added note and example to doc string

File: tests/air/two/test_enrich.py [hunk 4]
```diff
@@ -33,12 +34,12 @@ def test_join_to_lookup(
...
-def test_agg_CO2e_by_category_and_year(
+def test__agg_CO2e_by_category_and_year(
     transformed_joined: DataFrame,
     enriched: DataFrame,
 ) -> None:
     """CO2e is aggregated by category and EmissionsYear."""
-    returned: DataFrame = agg_CO2e_by_category_and_year(transformed_joined)
+    returned: DataFrame = _agg_CO2e_by_category_and_year(transformed_joined)
 
     expected: DataFrame = enriched
```
Comment (EFT-Defra) [inline tests/air/two/test_enrich.py:45]: Just updating the function names

File: src/oiflib/air/two/enrich.py [hunk 5]
```diff
@@ -36,16 +61,34 @@ def agg_CO2e_by_category_and_year(df: DataFrame) -> DataFrame:
...
+    _joined: DataFrame = _join_to_lookup(
+        df=df,
+        path_to_lookup=path_to_lookup,
+    )
+
+    _aggregated: DataFrame = _agg_CO2e_by_category_and_year(
+        df=_joined,
+    )
+
+    return _aggregated
```
Comment (EFT-Defra) [inline src/oiflib/air/two/enrich.py:94]: 1. Now setting the path to the lookup in this top level function as the default argument for the path_to_look parameter
2. Extended the doc string with references to the lower-level private functions it calls and an example of how to use it
3. Split the pipe into individual function calls to make the behaviour more explicit

**Review by TimAshelford — APPROVED** [review]
Thanks Ed, with your explanation this makes sense.

**Pull request merged**